dds/DCPS/NetworkResource.cpp

```diff
@@ -774,6 +797,15 @@ ACE_INET_Addr choose_single_coherent_address(const String& address, bool prefer_
   }
 #endif /* ACE_WIN32 */
 
+  size_t addr_count;
```

Comment:
I don't think you can embed the check for whether the resolved address belongs to a local interface within choose_single_coherent_address, because both the local and remote processes (potentially on a different host) will call choose_single_coherent_address. I think the check should get pulled out into a new helper function that takes the address list as a parameter (since get_fully_qualified_hostname already has the list) and an address to check against the list... and then get_fully_qualifed_hostname can use it to reject candidates from being the FQDN as well as to evaluate the potential of nonFQDN candidates.

Reply:
The reason I wanted to check the resolved addresses against the list of IP interfaces inside `choose_single_coherent_address` is that I think there can be cases where a hostname resolves to some valid IP and some incorrect IP, and `choose_single_coherent_address` may return an incorrect IP address instead of a correct one. In that case, we will ignore this hostname if we just rely on a single address returned by `choose_single_coherent_address` while actually, we can still use it with a correct IP mapped to it. So I wanted to do a more comprehensive check that checks all resolved addresses against the IP interfaces list. And `choose_single_coherent_address` seemed to be a good place to do that check. But I didn't know `choose_single_coherent_address` is also called by a remote peer to find the address of a local peer. So this won't work.

Reply:
From the remote peer's perspective, there isn't a way for it to know if an address returned from `choose_single_coherent_address` is a correct one. So I think we probably should just ignore the hostname with polluted IP addresses altogether.